WordPress/Sniffs/WP/OptionAutoloadSniff.php

```diff
@@ -0,0 +1,439 @@
+			// means it is not possible to determine the option names and autoload values passed to
+			// wp_set_option_autoload_values().
+			return;
+		}
+
+		$options = PassedParameters::getParameters( $this->phpcsFile, $array_token );
+
+		if ( empty( $options ) ) {
+			// Bail if the array is empty.
+			return;
+		}
+
+		foreach ( $options as $array_item ) {
+			$double_arrow_pointer = Arrays::getDoubleArrowPtr( $this->phpcsFile, $array_item['start'], $array_item['end'] );
+
+			if ( false === $double_arrow_pointer ) {
+				$start = $array_item['start'];
+				$clean = $array_item['clean'];
+			} else {
+				$array_item_parts = explode( '=>', $array_item['clean'] );
+				$start            = $double_arrow_pointer + 1;
+				$clean            = trim( $array_item_parts[1] );
+			}
+
+			$array_value_info = array(
+				'start' => $start,
+				'end'   => $array_item['end'],
+				'clean' => $clean,
+			);
+
+			$this->check_autoload_value( $array_value_info, 'wp_set_option_autoload_values' );
+		}
+	}
+
+	/**
+	 * Adds a PHPCS warning when autoload parameter is missing for the functions where
+	 * this parameter is optional.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param int    $stackPtr      The position of the current token in the stack.
+	 * @param string $function_name The token content (function name) which was matched
+	 *                              in lowercase.
+	 *
+	 * @return void
+	 */
+	private function maybe_display_missing_autoload_warning( $stackPtr, $function_name ) {
+		$this->phpcsFile->recordMetric( $stackPtr, self::METRIC_NAME, 'param missing' );
+
+		// Only display a warning for the functions in which the `$autoload` parameter is optional.
+		if ( isset( $this->autoload_is_optional[ $function_name ] ) ) {
+			$this->phpcsFile->addWarning(
+				'It is recommended to always pass the `$autoload` parameter when using %s() function.',
+				$stackPtr,
+				'Missing',
+				array( $this->tokens[ $stackPtr ]['content'] )
+			);
+		}
+	}
+
+	/**
+	 * Check the autoload value for possible violations.
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param array  $autoload_info Information about the autoload value (start and end tokens and
+	 *                              the clean value).
+	 * @param string $function_name The token content (function name) which was matched
+	 *                              in lowercase.
+	 *
+	 * @return void
+	 */
+	private function check_autoload_value( array $autoload_info, $function_name ) {
+		$param_first_token  = $this->phpcsFile->findNext(
+			Tokens::$emptyTokens,
+			$autoload_info['start'],
+			( $autoload_info['end'] + 1 ),
+			true
+		);
+		$param_second_token = $this->phpcsFile->findNext(
+			Tokens::$emptyTokens,
+			$param_first_token + 1,
+			( $autoload_info['end'] + 1 ),
+			true
+		);
+
+		$normalized_value = strtolower( $autoload_info['clean'] );
+
+		if ( \T_NS_SEPARATOR === $this->tokens[ $param_first_token ]['code'] && $param_second_token ) {
```

Comment:
Note: this needs to be changed when adding support for PHPCS 4.0. Not something that should be addressed in this PR.